ksqldb-rest-app/src/main/java/io/confluent/ksql/api/impl/QueryEndpoint.java

```diff
@@ -43,34 +44,38 @@
 import java.util.Objects;
 import java.util.Optional;
 import java.util.stream.Collectors;
-import org.apache.kafka.common.utils.Time;
 
 public class QueryEndpoint {
 
   private final KsqlEngine ksqlEngine;
   private final KsqlConfig ksqlConfig;
   private final PullQueryExecutor pullQueryExecutor;
+  private final Optional<PullQueryExecutorMetrics> pullQueryMetrics;
 
-  public QueryEndpoint(final KsqlEngine ksqlEngine, final KsqlConfig ksqlConfig,
-      final PullQueryExecutor pullQueryExecutor) {
+  public QueryEndpoint(
+      final KsqlEngine ksqlEngine,
+      final KsqlConfig ksqlConfig,
+      final PullQueryExecutor pullQueryExecutor,
+      final Optional<PullQueryExecutorMetrics> pullQueryMetrics
+  ) {
     this.ksqlEngine = ksqlEngine;
     this.ksqlConfig = ksqlConfig;
     this.pullQueryExecutor = pullQueryExecutor;
+    this.pullQueryMetrics = pullQueryMetrics;
   }
 
   public QueryPublisher createQueryPublisher(
       final String sql, final JsonObject properties,
       final Context context,
       final WorkerExecutor workerExecutor,
       final ServiceContext serviceContext) {
-    final long startTimeNanos = Time.SYSTEM.nanoseconds();
```

Comment:
This no longer measures the latency for this endpoint.  Why did you remove it?

Reply:
Yeah, you are right. I got a bit trigger happy with removing things. Will add it back